K2Bridge/Factories/AggregateFactory.cs

```diff
@@ -366,6 +364,68 @@ private static PercentileAggregate GetPercentileAggregate(string columnName, str
             return percentileAggregate;
         }
 
+        /// <summary>
+        /// Get tophits aggregate from a given <see cref="DataRow"/>.
+        /// </summary>
+        /// <param name="columnName">The column name.</param>
+        /// <param name="columnMetadata">The column metadata.</param>
+        /// <param name="row">The row to be parsed.</param>
+        /// <param name="logger">ILogger object for logging.</param>
+        /// <param name="query">QueryData containing query information.</param>
+        /// <returns><see cref="TopHitsAggregate"/>.</returns>
+        private static TopHitsAggregate GetTopHitsAggregate(string columnName, string[] columnMetadata, DataRow row, ILogger logger, QueryData query)
+        {
+            logger.LogTrace("Get TopHits aggregate for {}", columnName);
+
+            var random = new Random();
+            var topHitsAggregate = new TopHitsAggregate() { };
+
+            topHitsAggregate.Hits.SetTotal(0);
+
+            var rowValue = row[columnName];
+            if (rowValue.GetType() != typeof(System.DBNull))
+            {
+                var docCount = (long)row[BucketColumnNames.Count];
+                topHitsAggregate.Hits.SetTotal(docCount);
+
+                if (row[columnName] is JArray { HasValues: true } topHitsValues)
+                {
+                    var hits = new List<Hit>();
+
+                    foreach (JObject jObject in topHitsValues)
+                    {
+                        var hit = HitsFactory.Create(random.Next().ToString(), query.IndexName);
```

Comment:
are we passing in the query data just for the indexname?

Reply:
Yes, you prefer I pass only the index name ?